animations/shared-element-transitions/index.md

```diff
@@ -0,0 +1,684 @@
+   The destination route defines its widget tree at the end
+   of the transition.
+1. Trigger the animation by pushing the route on the Navigator's stack.
+   Popping a route from the Navigator’s stack triggers a return
+   transition.
+
+Flutter calculates the tweens required to transition the
+starting Hero to the ending Hero (interpolating size and
+position changes), and performs the animation in an overlay.
+
+The next section describes Flutter's process in greater detail.
+
+## Behind the scenes
+
+The source route contains the Hero inside an Inkwell.
+What happens when the InkWell receives a tap gesture?
+The following describes how Flutter performs the transition.
+
+<img src="images/hero-transition-0.png" alt="Before the transition the source hero appears in the source route">
+
+Before transition, the source Hero waits in the source route's widget
+tree. The destination route does not yet exist, and the overlay
+is blank and disabled.
+
+---
+
+<img src="images/hero-transition-1.png" alt="Transition begins and Destination Hero appears in the overlay">
+
+Tapping the Hero's InkWell triggers the animation, and the
+following occurs:
+
+* Offscreen, Flutter calculates the transition (including
+  position and size changes) from beginning to end,
+  moving the Hero along a curved path as described in the
+  Material motion spec. Flutter saves the size and location
+  of the Hero at the beginning and end of the transition.
+
+* Flutter places the destination Hero in the overlay, at the
+  same location and size as the _source_ Hero.
+
+* Flutter enables the overlay, obscuring the current route.
+
+* Flutter creates the destination route, placing it at the
+  top of the Navigator's stack (above the source route),
+  but it's obscured by the overlay.
+
+* The destination route won't contain the destination Hero
+  until the transition is complete.
+
+---
+
+<img src="images/hero-transition-2.png" alt="The Hero flies in the overlay to its final position and size">
+
+While the Hero is in-flight:
+
+* The same Animation object drives both
+  the route transition and the Hero animation.
+
+* As the Hero flies, its rectangular bounds are animated using
+  [Tween&lt;Rect&gt;,](https://docs.flutter.io/flutter/animation/Tween-class.html)
+  accessible via the Hero's `createRectTween` property.
+  By default, Flutter uses an instance of
+  [MaterialRectArcTween,](https://docs.flutter.io/flutter/material/MaterialRectArcTween-class.html)
+  which animates the rectangle's opposing corners along a curved path.
+  (See the [radial transformations](#radial-transformations) section
+  for an example that uses a different Tween animation.)
+
+---
+
+<img src="images/hero-transition-3.png" alt="When the transition is complete, the Hero is moved from the overlay to the destination route">
+
+When the flight completes:
+
+* Flutter moves the Hero widget from the overlay to the
+  destination route.
+
+* Flutter disables the (now blank) overlay.
```

Comment:
I wouldn't say this. We don't disable overlays and the diagram makes it clear that it's empty.